Library/ViewModels/LiveStreamEventDetailsViewModel.swift

```diff
@@ -0,0 +1,267 @@
+import KsApi
+import LiveStream
+import ReactiveSwift
+import ReactiveExtensions
+import Result
+import Prelude
+
+public protocol LiveStreamEventDetailsViewModelType {
+  var inputs: LiveStreamEventDetailsViewModelInputs { get }
+  var outputs: LiveStreamEventDetailsViewModelOutputs { get }
+}
+
+public protocol LiveStreamEventDetailsViewModelInputs {
+  func configureWith(project: Project, liveStream: Project.LiveStream, event: LiveStreamEvent?)
+  func liveStreamViewControllerStateChanged(state: LiveStreamViewControllerState)
+  func subscribeButtonTapped()
+  func setNumberOfPeopleWatching(numberOfPeople: Int)
+  func userSessionStarted()
+  func viewDidLoad()
+}
+
+public protocol LiveStreamEventDetailsViewModelOutputs {
+  var animateActivityIndicator: Signal<Bool, NoError> { get }
+  var animateSubscribeButtonActivityIndicator: Signal<Bool, NoError> { get }
+  var availableForText: Signal<String, NoError> { get }
+  var creatorAvatarUrl: Signal<URL?, NoError> { get }
+  var configureShareViewModel: Signal<(Project, LiveStreamEvent), NoError> { get }
+  var detailsStackViewHidden: Signal<Bool, NoError> { get }
+  var liveStreamTitle: Signal<String, NoError> { get }
+  var liveStreamParagraph: Signal<String, NoError> { get }
+  var numberOfPeopleWatchingText: Signal<String, NoError> { get }
+  var openLoginToutViewController: Signal<(), NoError> { get }
+  var retrievedLiveStreamEvent: Signal<LiveStreamEvent, NoError> { get }
+  var shareButtonEnabled: Signal<Bool, NoError> { get }
+  var showErrorAlert: Signal<String, NoError> { get }
+  var subscribeButtonAccessibilityHint: Signal<String, NoError> { get }
+  var subscribeButtonAccessibilityLabel: Signal<String, NoError> { get }
+  var subscribeButtonImage: Signal<String?, NoError> { get }
+  var subscribeButtonText: Signal<String, NoError> { get }
+  var subscribeLabelHidden: Signal<Bool, NoError> { get }
+  var subscribeLabelText: Signal<String, NoError> { get }
+}
+
+public final class LiveStreamEventDetailsViewModel: LiveStreamEventDetailsViewModelType,
+  LiveStreamEventDetailsViewModelInputs, LiveStreamEventDetailsViewModelOutputs {
+
+  //swiftlint:disable:next function_body_length
+  public init () {
+    let configData = Signal.combineLatest(
+      self.configData.signal.skipNil(),
+      self.viewDidLoadProperty.signal
+      )
+      .map(first)
+
+    let project = configData.map(first)
+
+    let eventEvent = configData
+      .switchMap { project, liveStream, optionalEvent in
+        fetchEvent(forProject: project, liveStream: liveStream, event: optionalEvent)
```

Comment:
do we want to add the scheduler to this? `.ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)`